cluster.go

```diff
@@ -995,8 +995,18 @@ func (c *Cluster) PinGet(h *cid.Cid) (api.Pin, error) {
 // the cluster.  Priority allocations are best effort.  If any priority peers
 // are unavailable then Pin will simply allocate from the rest of the cluster.
 func (c *Cluster) Pin(pin api.Pin) error {
-	_, err := c.pin(pin, []peer.ID{}, pin.Allocations)
-	return err
+	switch pin.Type {
```

Comment:
This switch/logic should probably go into the unexported `pin()`, which is what `repinFromPeer` calls (if I remember well). I'm pretty sure it's going to cause weird side effects otherwise.